app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,84 @@
+require 'uri'
+require 'net/http'
+require 'base64'
+
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is responsible for fetching JWK Set from JWKS-uri
+      class FetchJwksUriSigningKey < FetchSigningKeyInterface
+
+        def initialize(authentication_parameters:,
+          logger:,
+          fetch_required_secrets:,
+          resource_class:,
+          http_lib:,
+          create_jwks_from_http_response:)
+          @logger = logger
+
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+          @resource_id = authentication_parameters.authn_jwt_variable_id
+          @http_lib = http_lib
+          @create_jwks_from_http_response = create_jwks_from_http_response
+        end
+
+        def valid_configuration?
+          @valid_configuration ||= jwks_uri_resource_exists?
+        end
+
+        def fetch_signing_key
+          fetch_jwks_uri
+          fetch_jwks_keys
+        end
+
+        private
+
+        def jwks_uri_resource_exists?
+          !jwks_uri_resource.nil?
+        end
+
+        def jwks_uri_resource
+          @jwks_uri_resource ||= resource
+        end
+
+        def resource
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingJwtConfigurationValue.new(jwks_uri_resource_id))
+          @resource_class[jwks_uri_resource_id]
+        end
```

Comment:
this can be changed to:
```suggestion
        def jwks_uri_resource
          return @jwks_uri_resource if @jwks_uri_resource
          
 
 @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingJwtConfigurationValue.new(jwks_uri_resource_id))
          
          @jwks_uri_resource = @resource_class[jwks_uri_resource_id]
        end
```

Reply:
Done